### PR TITLE
Updating question/answer model & adding question dao

### PR DESCRIPTION
**Pull request by mattfreshwaters**
Updated Item model to include Many-to-one relationship with Question model.  Updated Question Model to include many-to-one relationship with the newly added answer model.  Getters/setters/attributes/hash/equals included in each model

### CODE REVIEW

**Review by jlavendtt — APPROVED** [review]

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Item.java [hunk 1]
```diff
@@ -21,6 +21,9 @@
     @OneToMany(mappedBy="item", cascade=CascadeType.ALL)
     private List<ProductPhoto> productPhotos;
 
+    @OneToMany(mappedBy = "question", cascade=CascadeType.ALL)
+    private List<Question> questions;
```
Comment (FeiYGH) [inline showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Item.java:25]: Hey Matt, just checking, I thought you said you were linking questions to ItemFamily. Let me know if you changed you mind

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Question.java [hunk 2]
```diff
@@ -21,6 +25,14 @@
 
     private LocalDate date;     // Store the date of when the question was asked
 
+    @JsonIgnore
+    @ManyToOne(fetch=FetchType.LAZY)
+    @NotBlank
+    private ItemFamily itemFamily;
+
+    @OneToMany(mappedBy = "answer", cascade=CascadeType.ALL)
```
Comment (FeiYGH) [inline showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Question.java:33]: mappedBy="question" -- mappedBy is what we use to tell Hibernate which variable we are using to represent parent class in our child class =)

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Answer.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    private Integer answerId;
+
+    private Integer userId;
+
+    private String answer;
+
+    private LocalDate date;
+
+    @JsonIgnore
+    @ManyToOne
```
Comment (FeiYGH) [inline showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Answer.java:24]: @ManyToOne(fetch=FetchType.LAZY)

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Item.java [hunk 4]
```diff
@@ -21,6 +21,9 @@
     @OneToMany(mappedBy="item", cascade=CascadeType.ALL)
     private List<ProductPhoto> productPhotos;
 
+    @OneToMany(mappedBy = "question", cascade=CascadeType.ALL)
```
Comment (FeiYGH) [inline showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/models/Item.java:24]: mappedBy="itemFamily" (that's what you named your variable in itemFamily where you did the ManyToOne relationship) -- mappedBy is what we use to tell Hibernate which variable we are using to represent parent class in our child class =)

**Review by FeiYGH — CHANGES_REQUESTED** [review]
Hey Matt take a look at my comments and let me know if you have any questions! Great work!!

**Review by FeiYGH — APPROVED** [review]
LGTM!

**Pull request merged**